pulsar-common/src/test/java/org/apache/pulsar/common/nar/NarUnpackerTest.java

```diff
@@ -118,6 +118,30 @@ public static void main(String[] args) {
         }
     }
 
+    @Test
+    void shouldReExtractWhenUnpackedDirectoryIsMissing() throws InterruptedException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        AtomicInteger exceptionCounter = new AtomicInteger();
+        AtomicInteger extractCounter = new AtomicInteger();
+
+        new Thread(() -> {
+            try {
+                File narWorkingDirectory = NarUnpacker.doUnpackNar(sampleZipFile, extractDirectory, extractCounter::incrementAndGet);
+                FileUtils.deleteFile(narWorkingDirectory, true);
+                NarUnpacker.doUnpackNar(sampleZipFile, extractDirectory, extractCounter::incrementAndGet);
+            } catch (Exception e) {
+                log.error("Unpacking failed", e);
+                exceptionCounter.incrementAndGet();
```

Comment:
This might introduce a flaky test since multiple threads are deleting the directory concurrently. It seems that a concurrency test doesn't make sense for testing that the unpacking happens when the working directory is missing.

Reply:
Running the code in a separate thread isn't necessary.

Reply:
Removed thread

Reply:
Please clean up the remaining parts that were no longer needed. I added specific comments.